### PR TITLE
Improve index for command line options.

### LINKED ISSUE #5592: Improve index for command line options
### Problem
Index entries created for command line options (the **cmdoption** and **option** directives) contains contain full line describing the option, with arguments and aliases. As result, similar options for different programs are not grouped, and separate entries for aliases are not created.

```
.. cmdoption:: --create <tarfile> <source1> ... <sourceN>
```
and
```
.. cmdoption:: --create <zipfile> <source1> ... <sourceN>
```
create two separate entries because the name of argument is different.

```
.. cmdoption:: -h, --help
```
creates a single entry for `-h, --help`, while
```
.. cmdoption:: -h
            --help
```
creates two entries, for `-h` and `--help`.

#### Procedure to reproduce the problem
Look for example at the index of Python documentation: https://docs.python.org/3/genindex-Symbols.html . It contains options for the Python interpreter itself and for some modules that provides a command line interface.

#### Error logs / results
```
--create <tarfile> <source1> ... <sourceN>
    tarfile command line option
--create <zipfile> <source1> ... <sourceN>
    zipfile command line option
```
```
--help
    command line option
    trace command line option
```
```
-c <command>
    command line option
-c <tarfile> <source1> ... <sourceN>
    tarfile command line option
-c <zipfile> <source1> ... <sourceN>
    zipfile command line option
-c, --catch
    unittest command line option
-c, --compress
    zipapp command line option
-c, --count
    trace command line option
```
```
-h
    command line option
-h, --help
    command line option
    timeit command line option
    tokenize command line option
    zipapp command line option
```
- https://docs.python.org/3/genindex-Symbols.html

#### Expected results
I expect all options with the same name are grouped and separate entries are created for aliases.
```
--create
    tarfile command line option
    zipfile command line option
```
```
--help
    command line option
    gzip command line option
    json.tool command line option
    timeit command line option
    tokenize command line option
    trace command line option
    zipapp command line option
```
```
-c
    command line option
    tarfile command line option
    trace command line option
    unittest command line option
    zipapp command line option
    zipfile command line option
```
```
-h
    command line option
    gzip command line option
    json.tool command line option
    timeit command line option
    tokenize command line option
    zipapp command line option
```

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Create separate entry for every alias of command line option. Not include option arguments in index, so that options with different names of arguments can be grouped.

Fixes #5592.

### CODE REVIEW

**Review by tk0miya — CHANGES_REQUESTED** [review]
Could you add testcase please?

File: sphinx/domains/std.py [hunk 1]
```diff
@@ -196,12 +196,18 @@ def add_target_and_index(self, firstname, sig, signode):
...
             # create only one index entry for the whole option
             if optname == firstname:
                 self.indexnode['entries'].append(
-                    ('pair', _('%scommand line option; %s') %
-                     ((currprogram and currprogram + ' ' or ''), sig),
+                    ('single', '%s; %s' % (descr, sig),
                      signode['ids'][0], '', None))
+            self.indexnode['entries'].append(
+                ('single', '%s; %s' % (optname, descr),
+                 signode['ids'][0], '', None))
```
Comment (tk0miya) [inline sphinx/domains/std.py:216]: Is this working fine for `-X <ARG>, -XXX <ARG>`?

**Comment by serhiy-storchaka** [comment]
I am not very well familiar with Sphinx testing. Could you please give me tips? In what file should new tests be added and what tests can be used as a pattern? Are there any existing tests for `cmdoption` or other directives that add index entries?

**Comment by tk0miya** [comment]
Sorry for late.

1. Please add directory named `tests/root/test-*` for new testcase
2. Please add a testcase for tests/test_*.py`. In this case, `tests/test_domain_std.py` would be better.
    - Please read other testcase to make your testcode.
    - In this case, I propose you 1) build HTML, 2) read `genindex.html` and 3) check it.

**Comment by tk0miya** [comment]
any updates? We'll call feature freeze in this weekend. So it would be nice to update this until then.

**Comment by tk0miya** [comment]
Merged #6494 instead. Thank you for proposal!

**Pull request closed without merging**